### DISCUSSION TITLE
Errors in sandbox test

### DISCUSSION THREAD

**Discussion by rusanpas** [saleweaver/python-amazon-sp-api#79] [Q&A]
Hello. First of all, thank you for your work with this project.

I am trying the tests and there are some that I have had to correct, but others do not work for me and give an error [{'message': 'Access to requested resource is denied.', 'Code': 'Unauthorized', 'details':' '}].

The sandbox tests that fail are:

test_get_inventory_summary_marketplace -> Access to requested resource is denied.
test_create_subscription -> Access to requested resource is denied.
test_delete_subscription -> Access to requested resource is denied.

The product and inventories tests, I had to fix them.

The credentials seem valid, since the rest of the methods work correctly. And in application roles, everything is marked.

On the other hand, I have developed the merchant_fullfillment model (which I would like to do pull request). But I can't get him to pass all the tests ...

The get_shipment is very simple and gives me [{'message': 'We encountered an internal error. Please try again. ',' Code ':' InternalFailure '}]

`
    @sp_endpoint ("/ mfn / v0 / shipments / {}")
    def get_shipment (self, shipment_id: str, ** kwargs) -> ApiResponse:
        """
        get_shipment (self, shipmentId: str) -> ApiResponse
        Returns a specified item and its attributes.

        ** Usage Plan: **

        ================================================== ==
        Rate (requests per second) Burst
        ================================================== ==
        eleven
        ================================================== ==

        For more information, see "Usage Plans and Rate Limits" in the Selling Partner API documentation.

        Args:
            shipment_id: str:

        Returns:
            GetShipmentResponse:
        """
        return self._request (fill_query_params (kwargs.pop ('path'), shipment_id), params = kwargs, add_marketplace = False)
`
Any suggestion?

Thanks

**Comment by saleweaver** [comment]
For your first problem, there's something wrong with your credentials. Note that there are 2 sections in seller central where you can assign permissions/request permissions from Amazon. Please contact seller support to fix the errors with your credentials. Every test passes with working credentials. 

For the second one, sandbox behavior is different from normal requests. Please look in the repo's "models" folder and find the sandbox-behavior parameters for the endpoint you want to test in the json file. Once you have that, you can test the endpoint. Thanks for contributing!

**Reply by rusanpas** [reply]
I'll check the permissions.

But regarding getshipment, I am already following the sandbox.

"request": {
                   "parameters": {
                     "shipmentId": {
                       "value": "abcddcba-00c3-4f6f-a63a-639f76ee9253"
                     }
                   }
                 },

```
def test_get_shipment ():
     res = MerchantFulfillment (). get_shipment ("abcddcba-00c3-4f6f-a63a-639f76ee9253")
     assert res.errors is None
     assert res.payload.get ('ShipmentId') == "abcddcba-00c3-4f6f-a63a-639f76ee9253"
```

Really, I don't know what else to try ...

**Reply by saleweaver** [reply]
Oh sorry, didn't see that. Everything you're doing looks correct, did you try to run the test with sandbox disabled, in a real shipment? I've had multiple sandbox endpoints that do not work, maybe this is one of them.

**Reply by rusanpas** [reply]
ok, I will ask for real data to test outside of sandbox

By the way, in this method getshipment is correct add_marketplace = false, right?

**Reply by saleweaver** [reply]
Yes, I think that's correct. I haven't worked with that endpoint before and don't have real data to test, but I would  have done it the same. 
If you want to, you can create a pull request with the current code so I can check this, maybe I see something while testing.

**Reply by rusanpas** [reply]
I can't do pull_request ... It returns 403, I don't know what I'm doing wrong ...
